src/utils/parse-text.js

```diff
@@ -28,29 +28,43 @@ const {
   siteDomains,
 } = CONFIG;
 
+const shortLinkRe = /\/[a-z\d-]{3,35}\/[\da-f]{6,10}(?:#[\da-f]{4,6})?/gi;
+const shortLinkExactRe = /^\/[a-z\d-]{3,35}\/[\da-f]{6,10}(?:#[\da-f]{4,6})?$/i;
+
 export class Link extends TLink {
   localDomains = [];
   hostname = null;
   path = '/';
   link;
+  isShort = false;
 
   constructor(link, localDomains) {
     super(link.offset, link.text);
     this.link = link;
     this.localDomains = localDomains;
+    this.isShort = shortLinkExactRe.test(link.text);
 
-    const m = this.link.href.match(/^https?:\/\/([^/]+)(.*)/i);
+    const m = this.href.match(/^https?:\/\/([^/]+)(.*)/i);
     if (m) {
       this.hostname = m[1].toLowerCase();
       this.path = m[2] || '/';
+    } else if (this.isShort) {
+      this.path = link.text;
     }
   }
 
-  get href() {
-    return this.link.href;
+  get pretty() {
+    if (this.isShort) {
+      return this.path;
+    }
+    return super.pretty;
   }
```

Comment:
Wait, I just noticed that in this last version we transform

```
https://freefeed.net/clbn/f0176a#3037
```

into

```
<a href="https://freefeed.net/clbn/f0176a#3037">https://freefeed.net/clbn/f0176a#3037</a>
```

instead of


```
<a href="https://freefeed.net/clbn/f0176a#3037">/clbn/f0176a#3037</a>
```

I thought we should have the latter one?

Reply:
I mean, certain long links should become short links (when not in edit mode), but now they don't.

Reply:
You're right, we wanted to shorten them according to the RFC. Fixed it.